frontend/lib/framework/Context.cpp

```diff
@@ -807,14 +840,61 @@ bool Context::queryCanUseSavedResultAndPushIfNot(
   return useSaved;
 }
 
+void Context::emitHiddenErrorsFor(const querydetail::QueryMapResultBase* result) {
+  CHPL_ASSERT(!result->emittedErrors);
+  for (auto& error : result->errors) {
+    reportError(this, error.get());
+  }
+  result->emittedErrors = true;
+  for (auto& dependency : result->dependencies) {
+    if (!dependency.query->emittedErrors && !dependency.errorCollectionRoot) {
+      emitHiddenErrorsFor(dependency.query);
+    }
+  }
+}
+
+static void findDependenciesForStoringErrors(const querydetail::QueryMapResultBase* result,
```

Comment:
could you add a comment about what this function is doing? I think I see but it wasn't obvious at a glance.

Anyway, what's the point of computing the dependencies and then visiting them in an unknown order? I would think it would be better for `storeErrorsFor` to be recursive and traverse the dependencies instead (i.e. combine the two functions). One issue with the current approach is that the `errorList` will have errors stored in a nondeterministic order. I suspect that will cause problems.